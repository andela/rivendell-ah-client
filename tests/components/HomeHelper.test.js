import { shallow, mount } from 'enzyme';
import React from 'react';
import { Router } from 'react-router-dom';
import {
  ArticleWithImg, Articles, PaginatedArticles, TopRated,
  Banner, FavoriteArticles, FavoriteArticleImg, CategoriesMenu
} from '../../src/components/helpers/HomeHelper';

const history = {
  location: {
    pathname: '/'
  },
  listen: () => {},
  createHref: () => {},
  push: () => {},
  replace: () => {},
};

describe('Testing the ArticleWithImg component', () => {
  it('Should render as expected', () => {
    const props = {
      article: {
        title: 'title',
        author: {
          username: 'username'
        }
      },
      backgroundImage: 'image',
      articleDescription: 'description',
      slideDown: () => {},
      slideUp: () => {},
      articleImgStyles: {}
    }
    const tree = shallow(<ArticleWithImg
      article={props.article}
      backgroundImage={props.backgroundImage}
      articleDescription={props.articleDescription}
      slideDown={props.slideDown}
      slideUp={props.slideUp}
      articleImgStyles={props.articleImgStyles}
    />);
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing the Articles component', () => {
  const props = {
    articles: [{
      slug: 'slug',
      title: 'title',
      description: 'description',
      author: {
        username: 'username'
      }
    }],
    slideDown: () => {},
    slideUp: () => {},
    articleImgStyles: {},
  };
  it('Should render a placeholder image if the article has no image', () => {
    const tree = shallow(<Articles
        articles={props.articles}
        slideDown={props.slideDown}
        slideUp={props.slideUp}
        articleImgStyles={props.articleImgStyles}
      />)
    expect(tree.contains(
      <ArticleWithImg
        article={props.articles[0]}
        backgroundImage="url(https://react.semantic-ui.com/images/wireframe/image.png)"
        articleDescription={props.articles[0].description}
        slideDown={props.slideDown}
        slideUp={props.slideUp}
        articleImgStyles={{}}
      />)).toBe(true);
  });
  it('Should render a the article\'s image if present', () => {
    props.articles[0].image = 'render-me';
    const tree = shallow(<Articles
        articles={props.articles}
        slideDown={props.slideDown}
        slideUp={props.slideUp}
        articleImgStyles={props.articleImgStyles}
      />)
    expect(tree.contains(
      <ArticleWithImg
        article={props.articles[0]}
        backgroundImage="url(render-me)"
        articleDescription={props.articles[0].description}
        slideDown={props.slideDown}
        slideUp={props.slideUp}
        articleImgStyles={{}}
      />)).toBe(true);
  });
  it('Should render \'...\' next to the 100th character of a description greater than 100', () => {
    props.articles[0].description = `Pellentesque ut neque. Donec posuere vulputate arcu.
    Vestibulum purus quam, scelerisque ut, mollis sed, nonummy id, metus. Nulla consequat
    massa quis enim. Nullam dictum felis eu pede mollis pretium.Praesent nec nisl a purus blandit
    viverra. Mauris sollicitudin fermentum libero. Etiam imperdiet imperdiet orci. Aliquam lobortis.`;
    const tree = shallow(<Articles
        articles={props.articles}
        slideDown={props.slideDown}
        slideUp={props.slideUp}
        articleImgStyles={props.articleImgStyles}
      />)
    expect(tree.contains(
      <ArticleWithImg
        article={props.articles[0]}
        backgroundImage="url(render-me)"
        articleDescription={`${props.articles[0].description.substring(0, 100)}...`}
        slideDown={props.slideDown}
        slideUp={props.slideUp}
        articleImgStyles={{}}
      />)).toBe(true);
  });
});

describe('Testing the PaginatedArticles component', () => {
  it('Should render as expected', () => {
    const props = {
      featArticlesActivePage: 1,
      feetArticlesPageCount: 2,
      feedPageCount: 22,
      feedActivePage: 12,
      token: 'token',
      displayArticles: [{
        slug: 'slug'
      }],
      articleImgStyles: {},
      slideDown: () => {},
      slideUp: () => {},
      handlePaginationChange: () => {},
    }
    let tree = PaginatedArticles({ ...props });
    tree = shallow(tree);
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing the TopRated component', () => {
  it('Should render as expected', () => {
    const props = {
      topRatedArticles: [{
        slug: 'slug',
      }]
    }
    let tree = TopRated({ ...props });
    tree = shallow(tree);
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing the Banner component', () => {
  it('Should render as expected', () => {
    let tree = Banner();
    tree = shallow(tree);
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing the FavoriteArticles component', () => {
  it('Should return null if the favoriteArticles array is empty', () => {
    const props = {
      favoriteArticles: [],
      nextArticle: {
        1: {
          display: 'none'
        }
      },
      getNextArticle: () => {},
    }
    let tree = FavoriteArticles({ ...props });
    expect(tree).toBeNull;
  });
  it('Should render as expected', () => {
    const props = {
      favoriteArticles: [{
        id: 'id',
        article: {
          slug: 'slug',
          title: 'title',
          author: {
            username: 'username',
          }
        }
      }],
      nextArticle: {
        1: {
          display: 'none'
        }
      },
      getNextArticle: () => {},
    }
    let tree = FavoriteArticles({ ...props });
    tree = shallow(tree);
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing the FavoriteArticleImg component', () => {
  it('Should render a placeholder image if no article image is present', () => {
    const favoriteArticle = {
      article: {
        image: null,
      }
    };
    const tree = shallow(<FavoriteArticleImg favoriteArticle={favoriteArticle} />);
    expect(tree.contains(
      <div
        id="article-with-img"
        style={{
          backgroundImage: 'url(https://react.semantic-ui.com/images/wireframe/image.png)',
          backgroundRepeat: 'no-repeat',
          backgroundSize: 'cover',
          backgroundPosition: 'center',
        }}
      />)).toBe(true);
  });
});

describe('Testing the CategoriesMenu component', () => {
  it('Should render as expected', () => {
    const categories = [{
      name: 'name',
      subcategories: [{
        id: 'id',
        name: 'name'
      }]
    }];
    const subcategoryClick = () => {};
    const tree = shallow(<CategoriesMenu subcategoryClick={subcategoryClick} categories={categories} />);
    expect(tree).toMatchSnapshot();
  });
});


