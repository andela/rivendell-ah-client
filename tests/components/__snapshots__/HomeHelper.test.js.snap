// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the ArticleWithImg component Should render as expected 1`] = `
<Link
  replace={false}
  to="/articles/undefined"
>
  <div
    id="article-with-img"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundImage": "image",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "width": "100%",
      }
    }
  >
    <div
      className="img-content"
      style={Object {}}
    >
      <div
        className="title-container"
      >
        <h4
          className="title"
        >
          title
        </h4>
      </div>
      <p
        className="desc"
      >
        description
      </p>
      <p
        className="author"
      >
        username
      </p>
      <p
        className="date"
      >
        Invalid Date
      </p>
    </div>
  </div>
</Link>
`;

exports[`Testing the Banner component Should render as expected 1`] = `
<div
  className="hero"
>
  <div
    className="text container"
  >
    <h1>
      AUTHORS HAVEN,
    </h1>
    <h1>
      A WHOLE NEW WORLD
    </h1>
    <Link
      replace={false}
      to="/signup"
    >
      <Button
        as="button"
        role="button"
      >
        Get started
      </Button>
    </Link>
  </div>
  <div
    className="hero-overlay"
  />
</div>
`;

exports[`Testing the CategoriesMenu component Should render as expected 1`] = `
<ul
  className="category-menu"
>
  <li
    key="0"
  >
    <ButtonToolbar
      bsClass="btn-toolbar"
    >
      <DropdownButton
        bsStyle="link"
        id="dropdown-no-caret"
        noCaret={true}
        title="name"
      >
        <MenuItem
          bsClass="dropdown"
          disabled={false}
          divider={false}
          eventKey="id"
          header={false}
          key="id"
          onClick={[Function]}
        >
          name
        </MenuItem>
      </DropdownButton>
    </ButtonToolbar>
  </li>
</ul>
`;

exports[`Testing the FavoriteArticles component Should render as expected 1`] = `
<div
  className="favorite-articles-container"
>
  <h3>
    Favorite articles
  </h3>
  <ul>
    <li
      className="favorite-article"
      key="id"
      style={Object {}}
    >
      <FavoriteArticleImg
        favoriteArticle={
          Object {
            "article": Object {
              "author": Object {
                "username": "username",
              },
              "slug": "slug",
              "title": "title",
            },
            "id": "id",
          }
        }
      />
      <div
        className="article-content"
      >
        <Link
          replace={false}
          to="/articles/slug"
        >
          <h4
            className="title"
          >
            title
          </h4>
          <p
            className="desc"
          />
          <p
            className="author"
          >
            username
          </p>
          <p
            className="date"
          >
            Invalid Date
          </p>
        </Link>
      </div>
      <Button
        as="button"
        className="next-article"
        onClick={[Function]}
        role="button"
      >
        Next article
      </Button>
    </li>
  </ul>
</div>
`;

exports[`Testing the PaginatedArticles component Should render as expected 1`] = `
<div
  className="articles"
>
  <h3>
    Feed
  </h3>
  <ul>
    <Articles
      articleImgStyles={Object {}}
      articles={
        Array [
          Object {
            "slug": "slug",
          },
        ]
      }
      slideDown={[Function]}
      slideUp={[Function]}
    />
  </ul>
  <div
    id="pagination-container"
  >
    <Pagination
      activePage={12}
      aria-label="Pagination Navigation"
      boundaryRange={1}
      className="pagination"
      ellipsisItem="..."
      firstItem={
        Object {
          "aria-label": "First item",
          "content": "«",
        }
      }
      lastItem={
        Object {
          "aria-label": "Last item",
          "content": "»",
        }
      }
      nextItem={
        Object {
          "aria-label": "Next item",
          "content": "⟩",
        }
      }
      onPageChange={[Function]}
      pageItem={Object {}}
      prevItem={
        Object {
          "aria-label": "Previous item",
          "content": "⟨",
        }
      }
      siblingRange={1}
      totalPages={22}
    />
  </div>
</div>
`;

exports[`Testing the TopRated component Should render as expected 1`] = `
<div
  className="top-rated"
>
  <h3>
    Top rated
  </h3>
  <ul>
    <li
      key="slug"
    >
      <Link
        replace={false}
        to="/articles/slug"
      >
        <span>
          <i
            className="material-icons"
          >
            star
          </i>
        </span>
        <div
          className="article-content"
        >
          <h4
            className="title"
          />
          <p
            className="desc"
          />
          <p
            className="author"
          />
          <p
            className="date"
          >
            Invalid Date
          </p>
        </div>
      </Link>
    </li>
  </ul>
</div>
`;
